x-pack/filebeat/input/gcs/metrics.go

```diff
@@ -0,0 +1,72 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package gcs
+
+import (
+	"github.com/rcrowley/go-metrics"
+
+	"github.com/elastic/beats/v7/libbeat/monitoring/inputmon"
+	"github.com/elastic/elastic-agent-libs/monitoring"
+	"github.com/elastic/elastic-agent-libs/monitoring/adapter"
+)
+
+// inputMetrics handles the input's metric reporting.
+type inputMetrics struct {
+	unregister        func()
+	url               *monitoring.String // URL of the input resource
+	errorsTotal       *monitoring.Uint   // number of errors encountered
+	decodeErrorsTotal *monitoring.Uint   // number of decode errors encountered
+
```

Comment:
Is is feasible to add a metric that indicates how many states (i.e. objects) are being tracked in the registry? And perhaps make the value be `-1` for stateless instances.

Reply:
Yea this should be possible to implement.